### ISSUE TITLE
Add fuzz tests for the cache merge logic.

### ISSUE THREAD

**Issue by bkolad** [Sovereign-Labs/first-read-last-write-cache#3]
It would be good to fuzz test the `CacheLog ::merge` method. We already have a programmatic way to crate test cases (see TestCase in tests)

**Comment by preston-evans98** [comment]
Good idea. Linking the [`proptest` library](https://docs.rs/proptest/latest/proptest/index.html) that we use throughout Sovereign's repos.

**Comment by mmchougule** [comment]
I am trying to work on this. I've got the protest set up to try first and last red values up to`std::u8::MAX`. The existing code without proptest does Read first, Read -> Write -> ReadThenWrite. Trying to fix a bug with using `proptest`:
Seems like the in `ReadThenWrite`, if a `Read` is followed by a `Write` with different values, the previous value needs to be discarded and only write should be saved.
@bkolad  Is that correct? Doesn't seem to be doing it but maybe there is another issue.
